initial version

IgorisB opened the pull request:
OpenID Connect client implementation
Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.OpenIdConnectModule_3.800.0-pr-1-0b87.zip

Commit:
Initial version

Commit:
initial version

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/VirtoCommerce/vc-module-openid-connect?pullRequest=1) <br/>All committers have signed the CLA.

Commit:
missing solution file

Commit:
fixing issues

ksavosteev commented on src/VirtoCommerce.OpenIdConnectModule.Web/webpack.config.js:
```diff
@@ -0,0 +1,52 @@
+const namespace = 'VirtoCommerce.OpenIdConnectModule'
```
this file can be removed

ksavosteev commented on src/VirtoCommerce.OpenIdConnectModule.Core/ModuleConstants.cs:
```diff
@@ -0,0 +1,70 @@
+    public const string OidcAuthenticationType = "oidc";
+    public const int ProviderPriority = 200;
+
+    public static class Security
```
class can be removed

ksavosteev commented on src/VirtoCommerce.OpenIdConnectModule.Core/ModuleConstants.cs:
```diff
@@ -0,0 +1,70 @@
+        }
+    }
+
+    public static class Settings
```
class can be removed

ksavosteev commented on src/VirtoCommerce.OpenIdConnectModule.Core/Models/OidcOptions.cs:
```diff
@@ -0,0 +1,134 @@
+        /// <summary>
+        /// Client Id 
+        /// </summary>
+        public string CallbackPath { get; set; } = "/signin-oidc";
```
It's better to use a different callback URLs for each external provider to avoid errors when unpacking the message
`/signin-oidc` is already used by default by AzureAD
https://josef.codes/unable-to-unprotect-the-message-state-asp-net-core/

ksavosteev commented on src/VirtoCommerce.OpenIdConnectModule.Web/Module.cs:
```diff
@@ -0,0 +1,99 @@
+
+                        openIdConnectOptions.ClaimActions.MapJsonKey(ClaimTypes.Name, "name");
+                        openIdConnectOptions.ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+                    });
```
add to support redirects for Front-end:
```cs
                            openIdConnectOptions.Events.OnRedirectToIdentityProvider = context =>
                            {
                                var oidcUrl = context.Properties.GetOidcUrl();
                                if (!string.IsNullOrEmpty(oidcUrl))
                                {
                                    context.ProtocolMessage.RedirectUri = oidcUrl;
                                }
                                return Task.CompletedTask;
                            };
```

Commit:
- removed webpack.config.js
- ModuleConstants.cs removed unused classes
- CallbackPath default value is removed
- Added support redirects for Front-end

Commit:
Merged with origin/initial

ksavosteev committed:
feat: refactoring

OlegoO committed:
Update README.md
